tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,217 @@
+    } else {
+        $vms->{$vm_name}->{dsturi}     = "xen+ssh://$dst_ip/";
+        $vms->{$vm_name}->{virsh_cmds} = ["virsh migrate --live", "xl migrate"];
+    }
+}
+
+#start a guest, a guest represents a vm for kvm test, two vms for xen test (a fv and a pv)
+sub start_guest {
+    my ($guest, $vms) = @_;
+    if (check_var('SYSTEM_ROLE', 'kvm')) {
+        create_vm($vms, $guest, "kvm", "x86_64");
+    } elsif (check_var('SYSTEM_ROLE', 'xen')) {
+        create_vm($vms, $guest, "fv", "x86_64");
+        create_vm($vms, $guest, "pv", "x86_64");
+    }
+}
+
+# perform live migration test and clean vm up after.
+sub migrate_test {
+    my ($vm_name, $vm_info) = @_;
+    for my $cmd (@{$vm_info->{virsh_cmds}}) {
+        #check if ip file is created. timeouts after 9 retries
+        assert_script_run("check_ip_file(){ for i in \$(seq 1 9);do if [ -f $images_path/guestip/$vm_name ]; then sleep 10; return; else sleep 10; fi; done; false; }; check_ip_file");
+        $vm_info->{ip} = script_output("cat $images_path/guestip/$vm_name");
+        if ("$cmd" eq "xl migrate") {
+            record_info("$cmd $vm_name $dst_ip");
+            assert_script_run("$cmd $vm_name $dst_ip");
+        } else {
+            record_info("$cmd $vm_name $vm_info->{dsturi}");
+            assert_script_run("$cmd $vm_name $vm_info->{dsturi}");
+        }
+        # visit web server running in vm check it's content
+        validate_script_output("curl -s http://$vm_info->{ip}/guest", sub { m/This is guest $vm_name/ });
+        #test vm disk write and read
+        assert_script_run("echo $vm_name > /tmp/test_vm_write");
+        assert_script_run("curl -s -k -T /tmp/test_vm_write -u bernhard:$testapi::password scp://$vm_info->{ip}/~/test_vm_write");
+        validate_script_output("curl -s -k -u bernhard:$testapi::password scp://$vm_info->{ip}/~/test_vm_write", sub { m/$vm_name/ });
+        assert_script_run("rm $images_path/guestip/$vm_name");
+        #shutdown migrated vm
+        if ($cmd eq "xl migrate") {
+            assert_script_run("ssh root\@$dst_ip xl shutdown $vm_name");
+        } else {
+            assert_script_run("ssh root\@$dst_ip virsh shutdown $vm_name");
+        }
+        #wait for shutdown complete
+        sleep 15;
+        #start again for next migration test
+        assert_script_run("virsh start $vm_name") if ($cmd ne $vm_info->{virsh_cmds}->[-1]);
+    }
+}
+
+sub cleanup {
+    my ($vm_name, $vm_info) = @_;
+    assert_script_run("virsh dumpxml $vm_name > $images_path/vm-configs/$vm_name.xml") unless ($vm_info->{reuse});
+    assert_script_run("virsh undefine $vm_name");
+}
+
+sub run {
+    select_console 'root-ssh';
+    $dst_ip = get_job_autoinst_vars(get_parents->[0])->{SUT_IP};
+    record_info("DST: $dst_ip");
+    record_info("SUT: $sut_ip");
+    assert_script_run("mount $sut_ip:/tmp/vm_images $images_path") unless get_var("NFS_PATH");
```

Comment:
Good question,
 It will be done by host autoyast.  You can find the profile in "Logs & Assets" in the each VR above.  The nfs section in profile is used to mount NFS_PATH.

Reply:
Thanks Tony. I have the same question and don't understand the answer. I've been looking in all autoyast xml files from http://10.67.183.130/tests/917#downloads and cannot find `NFS_PATH` there.

Reply:
You find three autoyast three profile there:
qem-host_12.xml
 sles-12-sp4-kvm.xml
 sles-12-sp5-kvm.xml

qem-host_12.xml is for the host. The others for Guests.
Click to open it with web browser. You will see the xml content at http://10.67.183.130/tests/917/file/qem-host_12.xml
As tests/917 is a 2-machine test so autoyast will set up nfs server on src host.  <nfs_server> section will do that.

tests/979 is 3-machines test. 
<nfs> section will mount NFS_PATH.

The host profile is generated from the template in the PR by prepare_profile test module which executed before boot_from_pxe

The guest profile is generated by create_profile subroutine.